mne/minimum_norm/tests/test_inverse.py

```diff
@@ -196,6 +203,54 @@ def test_make_inverse_operator():
 
 @slow_test
 @testing.requires_testing_data
+def test_inverse_operator_channel_ordering():
+    """Test MNE inverse computation is immune to channel reorderings
+    """
+    # These are with original ordering
+    evoked = _get_evoked()
+    noise_cov = read_cov(fname_cov)
+
+    fwd_orig = make_forward_solution(evoked.info, fname_trans, src_fname,
+                                     fname_bem, eeg=True, mindist=5.0)
+    fwd_orig = convert_forward_solution(fwd_orig, surf_ori=True)
+    inv_orig = make_inverse_operator(evoked.info, fwd_orig, noise_cov,
+                                     loose=0.2, depth=0.8,
+                                     limit_depth_chs=False)
+    stc_1 = apply_inverse(evoked, inv_orig, lambda2, "dSPM")
+
+    # Assume that a raw reordering applies to both evoked and noise_cov,
+    # so we don't need to create those from scratch. Just reorder them,
+    # then try to apply the original inverse operator
+    new_order = np.arange(len(evoked.info['ch_names']))
+    np.random.shuffle(new_order)
+    evoked.data = evoked.data[new_order]
+    evoked.info['ch_names'] = [evoked.info['ch_names'][n] for n in new_order]
+    evoked.info['chs'] = [evoked.info['chs'][n] for n in new_order]
+
+    cov_ch_reorder = [c for c in evoked.info['ch_names']
+                      if (c in noise_cov.ch_names)]
+
+    new_order_cov = [noise_cov.ch_names.index(name) for name in cov_ch_reorder]
+    noise_cov['data'] = noise_cov.data[np.ix_(new_order_cov, new_order_cov)]
+    noise_cov['names'] = [noise_cov['names'][idx] for idx in new_order_cov]
+
+    fwd_reorder = make_forward_solution(evoked.info, fname_trans, src_fname,
+                                        fname_bem, eeg=True, mindist=5.0)
+    fwd_reorder = convert_forward_solution(fwd_reorder, surf_ori=True)
+    inv_reorder = make_inverse_operator(evoked.info, fwd_reorder, noise_cov,
+                                        loose=0.2, depth=0.8,
+                                        limit_depth_chs=False)
+
+    stc_2 = apply_inverse(evoked, inv_reorder, lambda2, "dSPM")
+
+    assert_true(stc_1.subject == stc_2.subject)
+    assert_equal(stc_1.times, stc_2.times)
+    assert_allclose(stc_1.data, stc_2.data, rtol=1e-2, atol=1e-2)
```

Comment:
Yeah those copied tolerances needed to be a bit high because we were comparing the Python to the C code solutions. Here we are comparing things that *really* should be almost identical. I guess on further thought it might not end up being a perfect match, so just `assert_allclose` with `rtol=atol=1e-5` should be good enough.